app/src/main/kotlin/com/wire/android/services/OngoingCallService.kt

```diff
@@ -40,17 +54,35 @@ class OngoingCallService : Service() {
         if (userIdString != null && conversationIdString != null && callName != null) {
             val userId = qualifiedIdMapper.fromStringToQualifiedID(userIdString)
 
-            // we just need to keep some reference to the CallsScope,
-            // so it keeps CallManager alive
-            coreLogic.getSessionScope(userId).calls
+            scope.launch {
+                coreLogic.getSessionScope(userId)
+                    .calls
+                    .establishedCall()
+                    .collect { establishedCall ->
+                        if (establishedCall.isEmpty()) {
+                            appLogger.i("$TAG: stopSelf. Reason: call was ended")
+                            stopSelf()
+                        }
+                    }
+            }
 
             generateForegroundNotification(callName, conversationIdString, userIdString)
         } else {
-            stopForeground(true)
+            appLogger.i(
+                "$TAG: stopSelf. Reason: some of the parameter is absent. " +
+                        "userIdString: $userIdString, conversationIdString: $conversationIdString, callName: $callName"
+            )
```

Comment:
Should we upgrade the severity of these logs to at least `w`arning level?

Reply:
good point